homeassistant/components/fritzbox/climate.py

```diff
@@ -144,19 +175,20 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
     @property
     def preset_mode(self) -> str | None:
         """Return current preset mode."""
+        if self.data.holiday_active:
+            return PRESET_HOLIDAY
+        if self.data.summer_active:
+            return PRESET_SUMMER
         if self.data.target_temperature == self.data.comfort_temperature:
             return PRESET_COMFORT
         if self.data.target_temperature == self.data.eco_temperature:
             return PRESET_ECO
         return None
 
-    @property
-    def preset_modes(self) -> list[str]:
-        """Return supported preset modes."""
-        return [PRESET_ECO, PRESET_COMFORT]
-
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set preset mode."""
+        if self.data.holiday_active or self.data.summer_active:
```

Comment:
I take it, it's not possible to set holiday or summer via the integration?
In that case we should maybe raise an error here instead to tell the user they need to turn off holiday/summer via the app (or whatever) as it's not possible here.

Reply:
exactly - holiday and summer mode are configured via the device hub, but not the integration.
At the moment his PR includes code, which changes the list of available `hvac_modes` and `preset_modes` - based on the configured holiday or summer mode - to only each one available mode, so it is not possible to select any other hvac mode or preset for the user.
But we could change this to not changing the list of available hvac and presets, but raise an error to tell the user, that a change is not possible (_as you suggested_)

Reply:
after some thoughts about it, I think we should still change the list of available hvac modes and presets based on the holiday/summer mode, so the user can't select any unavailable hvac/preset